tests/testsuite/publish.rs

```diff
@@ -2173,18 +2161,17 @@ fn api_error_json() {
 
     p.cargo("publish --no-verify --registry alternative")
         .with_status(101)
-        .with_stderr(
-            "\
-[UPDATING] [..]
-[PACKAGING] foo v0.0.1 [..]
-[PACKAGED] [..] files, [..] ([..] compressed)
-[UPLOADING] foo v0.0.1 [..]
+        .with_stderr_data(str![[r#"
+[UPDATING] `alternative` index
+[PACKAGING] foo v0.0.1 ([ROOT]/foo)
+[PACKAGED] 3 files, [FILE_SIZE]B ([FILE_SIZE]B compressed)
+[UPLOADING] foo v0.0.1 ([ROOT]/foo)
 [ERROR] failed to publish to registry at http://127.0.0.1:[..]/
```

Comment:
`[ERROR] failed to publish to registry at http://127.0.0.1:64705/`

Could consider auto-redact the port to `[REGISTRY_PORT]`.